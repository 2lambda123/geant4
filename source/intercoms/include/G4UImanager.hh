//
// ********************************************************************
// * DISCLAIMER                                                       *
// *                                                                  *
// * The following disclaimer summarizes all the specific disclaimers *
// * of contributors to this software. The specific disclaimers,which *
// * govern, are listed with their locations in:                      *
// *   http://cern.ch/geant4/license                                  *
// *                                                                  *
// * Neither the authors of this software system, nor their employing *
// * institutes,nor the agencies providing financial support for this *
// * work  make  any representation or  warranty, express or implied, *
// * regarding  this  software system or assume any liability for its *
// * use.                                                             *
// *                                                                  *
// * This  code  implementation is the  intellectual property  of the *
// * GEANT4 collaboration.                                            *
// * By copying,  distributing  or modifying the Program (or any work *
// * based  on  the Program)  you indicate  your  acceptance of  this *
// * statement, and all its terms.                                    *
// ********************************************************************
//
//
// $Id: G4UImanager.hh,v 1.12 2001-10-02 00:32:06 asaim Exp $
// GEANT4 tag $Name: not supported by cvs2svn $
//

#ifndef G4UImanager_h
#define G4UImanager_h 1

#include "globals.hh"
#include "g4rw/ctoken.h"
//#include "g4rw/tvordvec.h"
#include "g4std/vector"
#include "g4std/fstream"
#include "G4VStateDependent.hh"
#include "G4UIcommandStatus.hh"
class G4UIcommandTree;
class G4UIcommand;
class G4UIsession;
class G4UIcontrolMessenger;
class G4UnitsMessenger;
class G4UIaliasList;

// class description:
//
//  This is a singlton class which controls the command manipulation
// and the user interface(s). The constructor of this class MUST NOT 
// invoked by the user.
//

class G4UImanager : public G4VStateDependent
{
  public: // with description
      static G4UImanager * GetUIpointer();
      //  A static method to get the pointer to the only existing object
      // of this class.

  protected:
      G4UImanager();
  public:
      ~G4UImanager();
  private:
      G4UImanager(const G4UImanager &right);
      const G4UImanager & operator=(const G4UImanager &right);
      int operator==(const G4UImanager &right) const;
      int operator!=(const G4UImanager &right) const;

  public: // with description
      G4String GetCurrentValues(const char * aCommand);
      //  This method returns a string which represents the current value(s)
      // of the parameter(s) of the specified command. Null string will be
      // returned if the given command is not defined or the command does
      // not support the GetCurrentValues() method.
      void AddNewCommand(G4UIcommand * newCommand);
      //  This method register a new command.
      void RemoveCommand(G4UIcommand * aCommand);
      //  This command remove the registered command. After invokation of this
      // command, that particular command cannot be applied.
      void ExecuteMacroFile(G4String fileName);
      //  A macro file defined by the argument will be read by G4UIbatch object.
      G4int ApplyCommand(const char * aCommand);
      G4int ApplyCommand(G4String aCommand);
      //  These two methods are identical. A command (and parameter(s)) given
      // by the method's argument will be applied. Zero will be returned in 
      // case the command is successfully executed. Positive non-zero value
      // will be returned if the command couldn't be executed. The meaning of
      // this non-zero value is the following.
      //   The returned number : xyy
      //        x00 : G4CommandStatus.hh enumeration
      //         yy : the problematic parameter (first found)
      void StoreHistory(G4String fileName="G4history.macro");
      void StoreHistory(G4bool historySwitch,
             G4String fileName="G4history.macro");
      //  The executed commands will be stored in the defined file. If 
      // "historySwitch" is false, saving will be suspended.
      void ListCommands(G4String direc);
      //  All commands registored under the given directory will be listed to
      // G4cout.
      void SetAlias(const char * aliasLine);
      void SetAlias(G4String aliasLine);
      //  Define an alias. The first word of "aliasLine" string is the
      // alias name and the remaining word(s) is(are) string value
      // to be aliased.
      void RemoveAlias(const char * aliasName);
      void RemoveAlias(G4String aliasName);
      //  Remove the defined alias.
      void ListAlias();
      //  Print all aliases.


  public:
      //void Foreach(G4String macroFile,G4String variableName,
      //             G4double initialValue,G4double finalValue,G4double stepSize);
      //void Foreach(G4String macroFile,G4String variableName,
      //             G4int initialValue,G4int finalValue,G4int stepSize);

      virtual G4bool Notify(G4ApplicationState requestedState);
      //  This method is exclusively invoked by G4StateManager and the user
      // must not use this method.
 
  private:
      void PauseSession(G4String msg);
      void CreateMessenger();
      G4UIcommandTree* FindDirectory(const char* dirName);

  public:
  // following three methods will be removed quite soon.
      void Interact();
      void Interact(const char * promptCharacters);
      void Interact(G4String promptCharacters);

  private:
      static G4UImanager * fUImanager;
      static G4bool fUImanagerHasBeenKilled;
      G4UIcommandTree * treeTop;
      G4UIsession * session;
      G4UIcontrolMessenger * UImessenger;
      G4UnitsMessenger * UnitsMessenger;
      G4String savedParameters;
      G4UIcommand * savedCommand;
      G4int verboseLevel;
      G4std::ofstream historyFile;
      G4bool saveHistory;
      G4std::vector<G4String> histVec;
      G4UIaliasList* aliasList;
      
      G4bool pauseAtBeginOfEvent;
      G4bool pauseAtEndOfEvent;


  public: // with description
      G4String GetCurrentStringValue(const char * aCommand, 
	    int parameterNumber=1, G4bool reGet=true);
      G4int GetCurrentIntValue(const char * aCommand, 
	    int parameterNumber=1, G4bool reGet=true);
      G4double GetCurrentDoubleValue(const char * aCommand,
	    int parameterNumber=1, G4bool reGet=true);
      G4String GetCurrentStringValue(const char * aCommand, 
	    const char * aParameterName, G4bool reGet=true);
      G4int GetCurrentIntValue(const char * aCommand, 
	    const char * aParameterName, G4bool reGet=true);
      G4double GetCurrentDoubleValue(const char * aCommand,
	    const char * aParameterName, G4bool reGet=true);
      //  These six methods returns the current value of a parameter of the
      // given command. For the first three methods, the ordering number of
      // the parameter (1 is the first parameter) can be given, whereas,
      // other three methods can give the parameter name.
      //  If "reGet" is true, actual request of returning the current value
      // will be sent to the corresponding messenger, while, if it is false,
      // the value stored in G4Umanager will be used. The later case is valid
      // for the sequential invokation for the same command.

      inline void SetPauseAtBeginOfEvent(G4bool vl)
      { pauseAtBeginOfEvent = vl; }
      inline G4bool GetPauseAtBeginOfEvent() const
      { return pauseAtBeginOfEvent; }
      inline void SetPauseAtEndOfEvent(G4bool vl)
      { pauseAtEndOfEvent = vl; }
      inline G4bool GetPauseAtEndOfEvent() const
      { return pauseAtEndOfEvent; }
      //  If the boolean flags are true, Pause() method of G4StateManager is invoked
      // at the very begining (before generating a G4Event object) or at the end of
      // each event. So that, in case a (G)UI session is defined, the user can interact.


  public:
      inline G4UIcommandTree * GetTree() const
      { return treeTop; }
      inline G4UIsession * GetSession() const
      { return session; }
  public: // with description
      inline void SetSession(G4UIsession *const value)
      { session = value; }
      //  This method defines the active (G)UI session.
     void SetCoutDestination(G4UIsession *const value);
     //  This method defines the destination of G4cout/G4cerr stream.
     // For usual cases, this method will be invoked by a concrete
     // (G)UI session class object and thus the user needs not to invoke
     // this method by him(her)self.

  public:
      inline void SetVerboseLevel(G4int val)
      { verboseLevel = val; }
      inline G4int GetVerboseLevel() const
      { return verboseLevel; }
      inline G4int GetNumberOfHistory() const
      { return histVec.size(); }
      inline G4String GetPreviousCommand(G4int i) const
      { 
        G4String st;
        if(i>=0 && i<G4int(histVec.size()))
        { st = histVec[i]; }
        return st;
      }

  // Old methods kept for backward compatibility
  //    inline G4UIcommandTree * GetTree() const
  //    { return treeTop; };
  //    inline G4UIsession * GetSession() const
  //    { return session; };
  //    inline void SetSession(G4UIsession *const  value)
  //    { session = value; };

};

#endif

