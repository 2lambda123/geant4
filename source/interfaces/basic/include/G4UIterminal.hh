//
// ********************************************************************
// * DISCLAIMER                                                       *
// *                                                                  *
// * The following disclaimer summarizes all the specific disclaimers *
// * of contributors to this software. The specific disclaimers,which *
// * govern, are listed with their locations in:                      *
// *   http://cern.ch/geant4/license                                  *
// *                                                                  *
// * Neither the authors of this software system, nor their employing *
// * institutes,nor the agencies providing financial support for this *
// * work  make  any representation or  warranty, express or implied, *
// * regarding  this  software system or assume any liability for its *
// * use.                                                             *
// *                                                                  *
// * This  code  implementation is the  intellectual property  of the *
// * GEANT4 collaboration.                                            *
// * By copying,  distributing  or modifying the Program (or any work *
// * based  on  the Program)  you indicate  your  acceptance of  this *
// * statement, and all its terms.                                    *
// ********************************************************************
//
//
// $Id: G4UIterminal.hh,v 1.8 2003-06-16 16:55:57 gunter Exp $
// GEANT4 tag $Name: not supported by cvs2svn $
//

#ifndef G4UIterminal_h
#define G4UIterminal_h 1

#include <fstream>
#include "G4UImanager.hh"
#include "G4VBasicShell.hh"
#include "G4VUIshell.hh"

//
//   Description:
//
//   This class inherits the class G4UIsession.
//   This is the class to use a character-terminal sesion.
//
//   Usage:  
//       G4UIsession* terminalSession = new G4UIterminal; 
//   or  G4UIsession* terminalSession = new G4UIterminal(new your-shell); 
//
//     A character-terminal session  "terminalSession" is instantiated.
//     G4cout stream is redirected by default to the constructed instance.
//
//   terminalSession-> SessionStart(); // "terminalSession" is started.
//   delete terminalSession;           // "terminalSession"  is deleted.
//
//
//   In default(no arguments are given), csh-like shell is instantiated.
//   If you want to use another shell (e.g. tcsh-like), you can give
//   your favorite shell in an argument of the constructor.
//
//   Which shell? / How to define your own shell?
//   Currently two kinds of shells,
//                   G4UIcsh / G4UItcsh
//   , are presented. 
//   They inherit the abstract base class, G4VUIshell.
//   In order to define your own shell, 
//     - Define your own shell class derived from G4VUIshell.
//     - Implement GetCommandLine() method (pure virtual).
//     - Add more functionality, if need.
//
//   For more detail, see source codes.
//

class G4UIterminal : public G4VBasicShell {
private:
  G4UImanager* UI;
  // shell
  G4VUIshell* shell;

  // program states
  G4bool iExit;
  G4bool iCont;

public:
  G4UIterminal(G4VUIshell* aShell=0);
  ~G4UIterminal();

  void SetPrompt(const G4String& prompt);

private:
  void ExecuteCommand(G4String aCommand);
  G4String GetCommand(const char* msg=0);
  G4bool GetHelpChoice(G4int& aInt);
  void ExitHelp();
  
public:
  // These methods are implementation of corresponding virtual methods
  // of G4UIsession class.
  virtual G4UIsession* SessionStart();  
  virtual void PauseSessionStart(G4String msg);
  virtual G4int ReceiveG4cout(G4String coutString);
  virtual G4int ReceiveG4cerr(G4String cerrString);

};

#endif

