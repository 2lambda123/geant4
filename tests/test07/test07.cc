// This code implementation is the intellectual property of
// the RD44 GEANT4 collaboration.
//
// By copying, distributing or modifying the Program (or any work
// based on the Program) you indicate your acceptance of this statement,
// and all its terms.
//
// $Id: test07.cc,v 1.1 1999-01-08 16:35:07 gunter Exp $
// GEANT4 tag $Name: not supported by cvs2svn $
//
// 
// --------------------------------------------------------------
//      GEANT 4 - exampleN02 
//
//      For information related to this code contact:
//      CERN, IT Division, ASD Group
// --------------------------------------------------------------
// Comments
//
// 
// --------------------------------------------------------------
#include "G4RunManager.hh"
#include "G4UImanager.hh"
#include "G4UIterminal.hh"

#ifdef G4VIS_USE
#include "T07VisManager.hh"
#endif

#include "T07DetectorConstruction.hh"
#include "T07PhysicsList.hh"
#include "T07PrimaryGeneratorAction.hh"
#include "T07RunAction.hh"
#include "T07EventAction.hh"
#include "T07SteppingAction.hh"

int main(int argc,char** argv) {

#ifdef MAKEHBOOK

  theHbookManager.SetFilename("test07.hbook");

#endif

  // Set the default random engine to RanecuEngine
  RanecuEngine defaultEngine;
  HepRandom::setTheEngine(&defaultEngine);

  // Construct the default run manager
  G4RunManager * runManager = new G4RunManager;

  // set mandatory initialization classes
  T07DetectorConstruction* detector = new T07DetectorConstruction;
  runManager->SetUserInitialization(detector);
  runManager->SetUserInitialization(new T07PhysicsList);
  
#if defined(G4VIS_USE) && !defined(MAKEHBOOK)
  // visualization manager
  G4VisManager* visManager = new T07VisManager;
  visManager->Initialize();
#endif
    
  // set user action classes
  runManager->SetUserAction(new T07PrimaryGeneratorAction(detector));
  runManager->SetUserAction(new T07RunAction);
  runManager->SetUserAction(new T07EventAction);
  runManager->SetUserAction(new T07SteppingAction);
  
  //Initialize G4 kernel
  runManager->Initialize();
    
  // get the pointer to the User Interface manager 
  G4UImanager* UI = G4UImanager::GetUIpointer();  

  if (argc==1)   // Define UI session for interactive mode.
    {
      // G4UIterminal is a (dumb) terminal.
      G4UIsession * session = new G4UIterminal;
      session->SessionStart();
      delete session;
    }
  else           // Batch mode
    { 
      G4String command = "/control/execute ";
      G4String fileName = argv[1];
      UI->ApplyCommand(command+fileName);
    }

  // job termination
#if defined(G4VIS_USE) && !defined(MAKEHBOOK)
  delete visManager;
#endif
  delete runManager;

  return 0;
}




