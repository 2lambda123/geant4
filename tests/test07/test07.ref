# Histos 4
# Bins 250
# Entries 1000
# Overflow 0
# Underflow 0
# Sum 1000
# Average 324.535
# RMS 63.4372
0 1
3 0
6 0
9 0
12 0
15 0
18 0
21 0
24 0
27 0
30 0
33 0
36 1
39 0
42 1
45 0
48 0
51 0
54 0
57 0
60 0
63 0
66 0
69 0
72 0
75 0
78 0
81 1
84 0
87 0
90 0
93 0
96 0
99 0
102 0
105 1
108 0
111 2
114 0
117 0
120 0
123 0
126 0
129 0
132 0
135 0
138 0
141 0
144 0
147 0
150 0
153 0
156 0
159 1
162 0
165 0
168 1
171 1
174 2
177 1
180 0
183 0
186 2
189 0
192 0
195 1
198 2
201 0
204 0
207 0
210 0
213 0
216 0
219 1
222 1
225 0
228 1
231 0
234 2
237 3
240 2
243 1
246 2
249 3
252 3
255 3
258 6
261 14
264 22
267 24
270 46
273 61
276 81
279 82
282 77
285 49
288 27
291 10
294 3
297 2
300 0
303 0
306 1
309 4
312 1
315 1
318 2
321 2
324 6
327 7
330 8
333 4
336 2
339 2
342 4
345 8
348 9
351 6
354 5
357 9
360 9
363 5
366 4
369 16
372 10
375 9
378 13
381 12
384 16
387 17
390 18
393 27
396 28
399 30
402 33
405 28
408 30
411 24
414 21
417 14
420 7
423 2
426 1
429 1
432 0
435 0
438 0
441 0
444 0
447 0
450 0
453 0
456 0
459 0
462 0
465 0
468 0
471 0
474 0
477 0
480 0
483 0
486 0
489 0
492 0
495 0
498 0
501 0
504 0
507 0
510 0
513 0
516 0
519 0
522 0
525 0
528 0
531 0
534 0
537 0
540 0
543 0
546 0
549 0
552 0
555 0
558 0
561 0
564 0
567 0
570 0
573 0
576 0
579 0
582 0
585 0
588 0
591 0
594 0
597 0
600 0
603 0
606 0
609 0
612 0
615 0
618 0
621 0
624 0
627 0
630 0
633 0
636 0
639 0
642 0
645 0
648 0
651 0
654 0
657 0
660 0
663 0
666 0
669 0
672 0
675 0
678 0
681 0
684 0
687 0
690 0
693 0
696 0
699 0
702 0
705 0
708 0
711 0
714 0
717 0
720 0
723 0
726 0
729 0
732 0
735 0
738 0
741 0
744 0
747 0
750 0


# Bins 100
# Entries 1000
# Overflow 6
# Underflow 0
# Sum 1000
# Average 44.7043
# RMS 34.5429
0 1
1 0
2 0
3 4
4 10
5 18
6 23
7 51
8 40
9 52
10 49
11 42
12 50
13 41
14 34
15 13
16 16
17 13
18 17
19 6
20 7
21 3
22 5
23 0
24 3
25 1
26 3
27 0
28 0
29 1
30 1
31 2
32 0
33 0
34 2
35 0
36 1
37 0
38 0
39 0
40 4
41 0
42 1
43 2
44 1
45 1
46 3
47 1
48 1
49 0
50 2
51 0
52 2
53 1
54 3
55 0
56 0
57 3
58 2
59 2
60 2
61 3
62 5
63 4
64 5
65 5
66 9
67 12
68 5
69 11
70 9
71 15
72 11
73 15
74 15
75 20
76 17
77 16
78 20
79 21
80 22
81 24
82 18
83 19
84 11
85 16
86 12
87 21
88 16
89 15
90 12
91 8
92 10
93 6
94 6
95 5
96 7
97 4
98 3
99 1
100 1


# Bins 100
# Entries 1000
# Overflow 480
# Underflow 0
# Sum 1000
# Average 489.144
# RMS 351.706
0 1
5 0
10 0
15 0
20 0
25 0
30 0
35 0
40 0
45 0
50 0
55 0
60 0
65 0
70 0
75 0
80 1
85 0
90 0
95 0
100 0
105 1
110 0
115 0
120 0
125 6
130 8
135 31
140 81
145 120
150 121
155 83
160 40
165 10
170 0
175 0
180 0
185 0
190 1
195 0
200 0
205 0
210 0
215 0
220 0
225 0
230 0
235 0
240 0
245 0
250 1
255 1
260 1
265 0
270 0
275 0
280 0
285 0
290 0
295 0
300 0
305 0
310 0
315 0
320 0
325 0
330 1
335 0
340 0
345 0
350 0
355 0
360 1
365 0
370 0
375 0
380 0
385 1
390 0
395 1
400 2
405 0
410 0
415 0
420 0
425 0
430 2
435 0
440 1
445 0
450 0
455 1
460 1
465 0
470 0
475 0
480 1
485 1
490 0
495 0
500 0


# Bins 100
# Entries 1000
# Overflow 496
# Underflow 0
# Sum 1000
# Average 1098.84
# RMS 1076.37
0 1
5 0
10 2
15 5
20 12
25 25
30 38
35 47
40 49
45 47
50 47
55 35
60 50
65 36
70 25
75 14
80 18
85 13
90 9
95 7
100 4
105 5
110 3
115 0
120 2
125 3
130 1
135 0
140 0
145 1
150 0
155 1
160 0
165 1
170 0
175 0
180 0
185 0
190 0
195 0
200 0
205 0
210 0
215 0
220 0
225 0
230 0
235 0
240 0
245 0
250 0
255 0
260 0
265 0
270 0
275 0
280 0
285 0
290 0
295 0
300 0
305 0
310 0
315 0
320 0
325 0
330 0
335 1
340 0
345 0
350 0
355 0
360 0
365 0
370 0
375 0
380 0
385 0
390 0
395 0
400 0
405 1
410 0
415 0
420 0
425 0
430 0
435 0
440 0
445 0
450 0
455 0
460 0
465 0
470 0
475 0
480 1
485 0
490 0
495 0
500 0
