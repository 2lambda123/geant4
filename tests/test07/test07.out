
***** Table : Nb of materials = 13 *****

 Material: Aluminium  density:  2.700  g/cm3  temperature: 273.15 K  pressure:   1.00 atm  RadLength:   8.893  cm
   --->  Element: Aluminium      Z = 13.0   N =  27.0   A =  26.98 g/mole  fractionMass: 100.00 %  Abundance 100.00 %

 Material: liquidArgon  density:  1.390  g/cm3  temperature: 273.15 K  pressure:   1.00 atm  RadLength:  14.065  cm
   --->  Element: liquidArgon      Z = 18.0   N =  40.0   A =  39.95 g/mole  fractionMass: 100.00 %  Abundance 100.00 %

 Material:   Copper  density:  8.960  g/cm3  temperature: 273.15 K  pressure:   1.00 atm  RadLength:   1.436  cm
   --->  Element:   Copper      Z = 29.0   N =  63.5   A =  63.55 g/mole  fractionMass: 100.00 %  Abundance 100.00 %

 Material:     Lead  density: 11.350  g/cm3  temperature: 273.15 K  pressure:   1.00 atm  RadLength:   5.612  mm
   --->  Element:     Lead      Z = 82.0   N = 207.2   A = 207.19 g/mole  fractionMass: 100.00 %  Abundance 100.00 %

 Material:    Water  density:  1.000  g/cm3  temperature: 273.15 K  pressure:   1.00 atm  RadLength:  36.092  cm
   --->  Element: Hydrogen  H   Z =  1.0   N =   1.0   A =   1.01 g/mole  fractionMass:  11.21 %  Abundance  66.67 %
   --->  Element:   Oxygen  O   Z =  8.0   N =  16.0   A =  16.00 g/mole  fractionMass:  88.79 %  Abundance  33.33 %

 Material: Scintillator  density:  1.032  g/cm3  temperature: 273.15 K  pressure:   1.00 atm  RadLength:  42.549  cm
   --->  Element:   Carbon  C   Z =  6.0   N =  12.0   A =  12.01 g/mole  fractionMass:  91.45 %  Abundance  47.37 %
   --->  Element: Hydrogen  H   Z =  1.0   N =   1.0   A =   1.01 g/mole  fractionMass:   8.55 %  Abundance  52.63 %

 Material:   quartz  density:  2.200  g/cm3  temperature: 273.15 K  pressure:   1.00 atm  RadLength:  12.295  cm
   --->  Element:  Silicon Si   Z = 14.0   N =  28.1   A =  28.09 g/mole  fractionMass:  46.75 %  Abundance  33.33 %
   --->  Element:   Oxygen  O   Z =  8.0   N =  16.0   A =  16.00 g/mole  fractionMass:  53.25 %  Abundance  66.67 %

 Material:      Air  density:  1.290 mg/cm3  temperature: 273.15 K  pressure:   1.00 atm  RadLength: 285.161   m
   --->  Element: Nitrogen  N   Z =  7.0   N =  14.0   A =  14.01 g/mole  fractionMass:  70.00 %  Abundance  72.71 %
   --->  Element:   Oxygen  O   Z =  8.0   N =  16.0   A =  16.00 g/mole  fractionMass:  30.00 %  Abundance  27.29 %

 Material:  Aerogel  density: 200.000 mg/cm3  temperature: 273.15 K  pressure:   1.00 atm  RadLength:   1.493   m
   --->  Element:  Silicon Si   Z = 14.0   N =  28.1   A =  28.09 g/mole  fractionMass:  29.22 %  Abundance  11.12 %
   --->  Element:   Oxygen  O   Z =  8.0   N =  16.0   A =  16.00 g/mole  fractionMass:  66.49 %  Abundance  44.42 %
   --->  Element: Hydrogen  H   Z =  1.0   N =   1.0   A =   1.01 g/mole  fractionMass:   4.19 %  Abundance  44.37 %
   --->  Element:   Carbon  C   Z =  6.0   N =  12.0   A =  12.01 g/mole  fractionMass:   0.10 %  Abundance   0.09 %

 Material: CarbonicGas  density: 27.000 mg/cm3  temperature: 325.00 K  pressure:  50.00 atm  RadLength:  13.406   m
   --->  Element:   Carbon  C   Z =  6.0   N =  12.0   A =  12.01 g/mole  fractionMass:  27.29 %  Abundance  33.33 %
   --->  Element:   Oxygen  O   Z =  8.0   N =  16.0   A =  16.00 g/mole  fractionMass:  72.71 %  Abundance  66.67 %

 Material: WaterSteam  density:  0.300 mg/cm3  temperature: 500.00 K  pressure:   2.00 atm  RadLength:   1.203  km
   --->  Element: Hydrogen  H   Z =  1.0   N =   1.0   A =   1.01 g/mole  fractionMass:  11.21 %  Abundance  66.67 %
   --->  Element:   Oxygen  O   Z =  8.0   N =  16.0   A =  16.00 g/mole  fractionMass:  88.79 %  Abundance  33.33 %

 Material: Galactic  density:  0.000 mg/cm3  temperature:   0.10 K  pressure:   0.00 atm  RadLength: 6317232936888469300000.000  km
   --->  Element: Galactic      Z =  1.0   N =   1.0   A =   1.01 g/mole  fractionMass: 100.00 %  Abundance 100.00 %

 Material:     Beam  density:  0.010 mg/cm3  temperature: 273.15 K  pressure:   0.02 atm  RadLength:  36.786  km
   --->  Element: Nitrogen  N   Z =  7.0   N =  14.0   A =  14.01 g/mole  fractionMass:  70.00 %  Abundance  72.71 %
   --->  Element:   Oxygen  O   Z =  8.0   N =  16.0   A =  16.00 g/mole  fractionMass:  30.00 %  Abundance  27.29 %



 The calorimeter is made of 10 layers of: [ 10mm of Lead + 5mm of liquidArgon ] 
T07PhysicsList::SetCuts:CutLength : 2 (mm)

phot:  Total cross sections from a parametrisation(L.Urban). Good description from 10 KeV to 50 MeV for all Z
        PhysicsTables from 10 keV to 50 MeV in 100 bins. 

compt:  Total cross sections from a parametrisation(L.Urban). Good description from 10 KeV to (100/Z) GeV. 
        Scattered gamma energy according Klein-Nishina.
        PhysicsTables from 10 keV to 1e+02 GeV in 100 bins. 

conv:  Total cross sections from a parametrisation(L.Urban). Good description from 1.5 MeV to 100 GeV for all Z. 
        e+e- energies according Bethe-Heitler
        PhysicsTables from 1 MeV to 1e+02 GeV in 100 bins. 

msc:   Tables of transport mean free paths.
          New model of MSC , computes the lateral 
          displacement of the particle , too.
        PhysicsTables from 1e+02  eV to 1e+02 TeV in 100 bins. 

eIoni:  delta cross sections from Moller+Bhabha. Good description from 1 KeV to 100 GeV.
        delta ray energy sampled from  differential Xsection.
        PhysicsTables from 1 keV to 1e+02 TeV in 100 bins. 

eBrem:  Total cross sections from a parametrisation(L.Urban). Good description from 1 KeV to 100 GeV.
        log scale extrapolation above 100 GeV 
        Gamma energy sampled from a parametrised formula.
        PhysicsTables from 1 keV to 1e+02 TeV in 100 bins. 

annihil:  Total cross section from Heilter formula (annihilation into 2 photons).
        gamma energies sampled according Heitler
        PhysicsTables from 10 keV to 10 TeV in 100 bins. 

msc:   Tables of transport mean free paths.
          New model of MSC , computes the lateral 
          displacement of the particle , too.
        PhysicsTables from 1e+02  eV to 1e+02 TeV in 100 bins. 

hIoni:    Knock-on electron cross sections . 
         Good description above the mean excitation energy.
         delta ray energy sampled from  differential Xsection.
        PhysicsTables from 1 keV to 1e+02 TeV in 100 bins. 

msc:   Tables of transport mean free paths.
          New model of MSC , computes the lateral 
          displacement of the particle , too.
        PhysicsTables from 1e+02  eV to 1e+02 TeV in 100 bins. 

MuIoni:  knock-on electron cross sections .
          Good description above the mean excitation energy.
         delta ray energy sampled from  differential Xsection.
      PhysicsTables from 1 keV to 1e+03 PeV in 100 bins. 

MuBrems:  cross sections from R. Kokoulin 
          Good description up to 1000 TeV.
    PhysicsTables from 1 GeV to 1e+03 PeV in 100 bins. 

MuPairProd:  cross sections from R. Kokoulin 
          Good description up to 1000 TeV.
    PhysicsTables from 1 GeV to 1e+03 PeV in 50 bins. 
Idle> Idle> /control/saveHistory
Idle> /run/verbose 2
Idle> #
Idle> /run/particle/dumpCutValues
============= The cut Energy ==============================
                          gamma          e-         mu-      proton     neutron
Cut in range [mm]             2           2           2           2           2
Cut in energy [KeV]
          Aluminium         9.6       1e+03     7.4e+03     1.9e+04        0.99
        liquidArgon         8.6     5.6e+02     4.8e+03     1.2e+04        0.99
             Copper          36     2.7e+03     1.3e+04     3.4e+04        0.99
               Lead     1.2e+02     2.6e+03     1.3e+04     3.2e+04        0.99
              Water         3.9     5.7e+02       5e+03     1.3e+04        0.99
       Scintillator         3.1     5.8e+02       5e+03     1.3e+04        0.99
             quartz         7.6     9.1e+02     6.9e+03     1.8e+04        0.99
                Air        0.99        0.99          82     1.7e+02        0.99
            Aerogel         2.3     1.8e+02     1.9e+03     4.8e+03        0.99
        CarbonicGas        0.99          54       6e+02     1.3e+03        0.99
         WaterSteam        0.99        0.99          30          45        0.99
           Galactic        0.99        0.99        0.99        0.99        0.99
               Beam        0.99        0.99        0.99        0.99        0.99
===================================================
Idle> #
Idle> #
Idle> /gun/particle e+
Idle> /gun/energy 300 MeV
Idle> /run/beamOn 500
### Run 0 start.
Start closing geometry.
Start Run processing.
Run terminated.
Run Summary
  Number of events processed : 500
  User=54s Real=59s Sys=5.7s
Idle> /calor/setAbsMat Aluminium

 The calorimeter is made of 10 layers of: [ 10mm of Aluminium + 5mm of liquidArgon ] 
Idle> /calor/setGapMat Aerogel

 The calorimeter is made of 10 layers of: [ 10mm of Aluminium + 5mm of Aerogel ] 
Idle> /calor/setAbsThick 2. cm

 The calorimeter is made of 10 layers of: [ 20mm of Aluminium + 5mm of Aerogel ] 
Idle> /calor/setGapThick 5. cm

 The calorimeter is made of 10 layers of: [ 20mm of Aluminium + 50mm of Aerogel ] 
Idle> /calor/setNbOfLayers 30

 The calorimeter is made of 30 layers of: [ 20mm of Aluminium + 50mm of Aerogel ] 
Idle> /calor/setSizeYZ   1.5 m

 The calorimeter is made of 30 layers of: [ 20mm of Aluminium + 50mm of Aerogel ] 
Idle> /calor/update

 The calorimeter is made of 30 layers of: [ 20mm of Aluminium + 50mm of Aerogel ] 
Idle> /gun/particle gamma
Idle> /gun/energy 500 MeV
Idle> /run/beamOn 500
### Run 1 start.
Start closing geometry.
Start Run processing.
Run terminated.
Run Summary
  Number of events processed : 500
  User=1.5e+02s Real=1.6e+02s Sys=15s
Idle> G4 kernel has come to Quit state.
Deletion of G4 kernel class start.
UserDetectorConstruction deleted.
UserPhysicsList deleted.
UserRunAction deleted.
UserPrimaryGenerator deleted.
G4SDManager deleted.
EventManager deleted.
UImanager deleted.
StateManager deleted.
RunManager is deleting.
